pennylane/wires.py

```diff
@@ -196,6 +196,9 @@ def map(self, wire_map):
             wire_map (dict): Dictionary containing all wire labels used in this object as keys, and unique
                              new labels as their values
         """
+        # Make sure wire_map has `Wires` keys and values so that the `in` operator always works
+        wire_map = {Wires(k): Wires(v) for k, v in wire_map.items()}
```

Comment:
After reading this PR I was wondering if we should think of a wire map just as dictionaries of labels, not `Wires` objects, and rather search for the labels in the self.labels here. Would that be more elegant? 

The only difference is that using Wires objects ensures that labels are valid types at all times...But this usually gets checked anyways (i.e. if the `new_wires` gets converted to `Wires`)...

Reply:
That's my first thought as well. But I ended up with enforcing `wire_map` items to be `Wires` because it's a single line change and worry free when the awesome `Wires` class does all the heavy lifting 😀.